Dumper/TranslationDumper.php

```diff
@@ -207,14 +207,33 @@ private function dumpTranslationsPerDomain($pattern, array $formats, $target)
         }
     }
 
-    private function dumpTranslationsPerLocale($pattern, array $formats, $target)
+    private function dumpTranslationsPerLocale($pattern, array $formats, $target, $mergeFallback)
     {
-        foreach ($this->getTranslations() as $locale => $domains) {
+        $allTranslations = $this->getTranslations();
+
+        foreach ($allTranslations as $locale => $domains) {
+            $translations = array($locale => $domains);
+
+            if ($mergeFallback) {
+                $localesFallback = array_unique(array(current(explode('_', $locale)), $this->localeFallback));
```

Comment:
that's broken. You can have multiple `_` in a locale, which defines multiple levels of fallbacks.